internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -303,6 +303,9 @@ func (in *RunOptions) generateRunArguments() []string {
 
 // Run runs a Docker container with the sepcified options.
 func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
+	type exitCodeError interface {
+		ExitCode() int
+	}
```

Comment:
Why do we define an interface here instead of checking if it is `ErrContainerExited`?